File: src/main/java/com/ncc/gaia/OntologyMapping.java
package com.ncc.gaia;

import org.apache.jena.rdf.model.Resource;

import java.util.Set;

/**
 * A domain ontology.
 */
interface OntologyMapping {
    String NAMESPACE;

    Set<String> entityShortNames();
    Resource entityType(String ontology_type);

    Resource relationType(String relationName);
    Resource eventType(String eventName);
    Resource eventArgumentType(String argName);
    /**
     * Given a relation, get its two argument types in the same order as in LDC annotation.
     */
    ResourcePair relationArgumentTypes(Resource relation);

    /**
     * Is an object of this type allowed to have a name property?
     */
    boolean typeAllowedToHaveAName(Resource type);
    boolean typeAllowedToHaveTextValue(Resource type);
    boolean typeAllowedToHaveNumericValue(Resource type);
}

File: src/main/java/edu/isi/gaia/OntologyMapping.kt
package edu.isi.gaia

import org.apache.jena.rdf.model.Resource

/**
 * Specifies how ColdStart type strings should be mapped to RDF IRIs.
 */
interface OntologyMapping {
    fun prefixes(): Map<Resource, String>
    fun entityType(ontology_type: String): Resource?
    fun relationType(relationName: String): Resource?
    fun eventType(eventName: String): Resource?
    fun eventArgumentType(argName: String): Resource?
    fun relationArgumentType(relationArgumentName: String): Resource?

    fun typeAllowedToHaveAName(type: Resource): Boolean
    fun typeAllowedToHaveTextValue(type: Resource): Boolean
    fun typeAllowedToHaveNumericValue(type: Resource): Boolean
}
